src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlHStoreKeysTranslator.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.EntityFrameworkCore.Query.Expressions;
+using Microsoft.EntityFrameworkCore.Query.Expressions.Internal;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.EntityFrameworkCore.Query.ExpressionTranslators.Internal
+{
+    public class NpgsqlHStoreKeysTranslator : IMethodCallTranslator
+    {
+        public Expression Translate(MethodCallExpression methodCallExpression)
+        {
+            if(methodCallExpression.Method == 
+                typeof(NpgsqlDbFunctionsExtensions)
```

Comment:
store in static field to avoid doing reflection for every translation.
Same for the other If